From 33ee9624d4fb50f29f2e3c8ffd0c7efa95a03f02 Mon Sep 17 00:00:00 2001
From: Dimitri John Ledkov <dimitri.j.ledkov@intel.com>
Date: Tue, 17 Mar 2015 16:37:07 +0100
Subject: [PATCH] resolved: Do not add .busname dependencies, when compiling
 without kdbus.

(cherry picked from commit defa8e675b2903ad53e093bb2847c7256f0779a5)

Cherry-picked from: 33ff64c
Resolves: #1222517
---
 Makefile.am                                                           | 2 +-
 units/.gitignore                                                      | 1 +
 units/{systemd-resolved.service.in => systemd-resolved.service.m4.in} | 2 ++
 3 files changed, 4 insertions(+), 1 deletion(-)
 rename units/{systemd-resolved.service.in => systemd-resolved.service.m4.in} (96%)

diff --git a/Makefile.am b/Makefile.am
index 8474b29..604eaf2 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -5520,7 +5520,7 @@ dist_dbussystemservice_DATA += \
 	src/resolve/org.freedesktop.resolve1.service
 
 EXTRA_DIST += \
-	units/systemd-resolved.service.in
+	units/systemd-resolved.service.m4.in
 
 SYSTEM_UNIT_ALIASES += \
 	systemd-resolved.service dbus-org.freedesktop.resolve1.service
diff --git a/units/.gitignore b/units/.gitignore
index 638a7ab..7f3e0d0 100644
--- a/units/.gitignore
+++ b/units/.gitignore
@@ -58,6 +58,7 @@
 /systemd-reboot.service
 /systemd-remount-fs.service
 /systemd-resolved.service
+/systemd-resolved.service.m4
 /systemd-hibernate-resume@.service
 /systemd-rfkill@.service
 /systemd-shutdownd.service
diff --git a/units/systemd-resolved.service.in b/units/systemd-resolved.service.m4.in
similarity index 96%
rename from units/systemd-resolved.service.in
rename to units/systemd-resolved.service.m4.in
index b643da9..d133847 100644
--- a/units/systemd-resolved.service.in
+++ b/units/systemd-resolved.service.m4.in
@@ -10,11 +10,13 @@ Description=Network Name Resolution
 Documentation=man:systemd-resolved.service(8)
 After=systemd-networkd.service network.service
 
+m4_ifdef(`ENABLE_KDBUS',
 # On kdbus systems we pull in the busname explicitly, because it
 # carries policy that allows the daemon to acquire its name.
 Wants=org.freedesktop.resolve1.busname
 After=org.freedesktop.resolve1.busname
 
+)m4_dnl
 [Service]
 Type=notify
 Restart=always
